alica_engine/include/engine/collections/RobotProperties.h

```diff
@@ -38,9 +38,8 @@ class RobotProperties {
 protected:
     const supplementary::AgentID* agentId;
 
-    string defaultRole;
-    map<string, Characteristic*> characteristics;
-    map<long, Capability*> capabilities;
+    std::string defaultRole;
+    std::map<std::string, const Characteristic*> characteristics;
```

Comment:
characteristics are only read once from the configuration file, so ideally this should be const. The corresponding references to characteristics would need to be changed in **RobotProperties::readFromConfig**.

Unless this is forward thinking in the event robots are able to acquire new capabilities dynamically if we want to enable ALICA for use in real Transformers...

Reply:
maybe next time....